lucet-concurrency-tests/tests/killswitch.rs

```diff
@@ -0,0 +1,731 @@
+        }
+    }
+
+    extern "C" fn hit_sigstack_guard_page(vmctx: *mut lucet_vmctx) {
+        extern "C" {
+            fn lucet_vmctx_get_globals(vmctx: *mut lucet_vmctx) -> *mut u8;
+        }
+
+        unsafe {
+            let globals_base = lucet_vmctx_get_globals(vmctx);
+
+            // Using the default limits, the globals are a page; try to write just off the end
+            *globals_base.offset(0x1000) = 0;
+        }
+    }
+
+    extern "C" fn do_nothing(_vmctx: *mut lucet_vmctx) -> () {}
+
+    extern "C" fn run_hostcall(vmctx: *mut lucet_vmctx) -> bool {
+        extern "C" {
+            fn real_hostcall(vmctx: *mut lucet_vmctx) -> bool;
+        }
+        unsafe { real_hostcall(vmctx) }
+    }
+
+    extern "C" fn run_yielding_hostcall(vmctx: *mut lucet_vmctx) -> () {
+        extern "C" {
+            fn yielding_hostcall(vmctx: *mut lucet_vmctx) -> ();
+        }
+        unsafe { yielding_hostcall(vmctx) }
+    }
+
+    MockModuleBuilder::new()
+        .with_export_func(MockExportBuilder::new(
+            "onetwothree",
+            FunctionPointer::from_usize(onetwothree as usize),
+        ))
+        .with_export_func(MockExportBuilder::new(
+            "infinite_loop",
+            FunctionPointer::from_usize(infinite_loop as usize),
+        ))
+        .with_export_func(MockExportBuilder::new(
+            "run_guest",
+            FunctionPointer::from_usize(run_guest as usize),
+        ))
+        .with_export_func(MockExportBuilder::new(
+            "do_nothing",
+            FunctionPointer::from_usize(do_nothing as usize),
+        ))
+        .with_export_func(MockExportBuilder::new(
+            "run_hostcall",
+            FunctionPointer::from_usize(run_hostcall as usize),
+        ))
+        .with_export_func(MockExportBuilder::new(
+            "run_yielding_hostcall",
+            FunctionPointer::from_usize(run_yielding_hostcall as usize),
+        ))
+        .with_export_func(MockExportBuilder::new(
+            "fatal",
+            FunctionPointer::from_usize(fatal as usize),
+        ))
+        .with_export_func(MockExportBuilder::new(
+            "hit_sigstack_guard_page",
+            FunctionPointer::from_usize(hit_sigstack_guard_page as usize),
+        ))
+        .build()
+}
+
+/// This test hostcall just needs to exist so that we can call it. `LockTestpoints` provide points
+/// for us to ensure that tests that need to happen inside a hostcall, will happen inside a
+/// hostcall.
+#[lucet_hostcall]
+#[no_mangle]
+pub fn real_hostcall(_vmctx: &mut Vmctx) -> bool {
+    true
+}
+
+/// This test hostcall will immediately yield. This is used to test termination of a
+/// yielded instance.
+#[lucet_hostcall]
+#[no_mangle]
+pub fn yielding_hostcall(vmctx: &mut Vmctx) {
+    vmctx.yield_();
+}
+
+pub fn test_c_with_instrumented_guest_entry<F, R>(dir: &str, cfile: &str, f: F) -> R
+where
+    F: FnOnce(InstanceHandle) -> R,
+{
+    test_ex(|| {
+        unsafe {
```

Comment:
Asking from a place of curiosity and wanting to learn more, why is this `unsafe`?

Reply:
Modifying a `static mut` is `unsafe` because there is no language-provided protections against race conditions. https://doc.rust-lang.org/reference/items/static-items.html#mutable-statics